### PR TITLE
Don 5 wishcard tests

### PR DESCRIPTION
**Pull request by Jacobjeevan**
- Added wishcard tests
- Some other changes were made to make testing smoother
- Also wrapped logic for multer/aws setup in a check for environment (only setup for production; I had used one of my buckets initially, but I think this is a better setup than adding S3/AWS credentials into test env file).

### CODE REVIEW

File: server/routes/wishCards.js [hunk 1]
```diff
@@ -27,10 +42,26 @@ const {
...
-const s3 = new AWS.S3({
-  accessKeyId: process.env.AWS_KEY,
-  secretAccessKey: process.env.AWS_SECRET,
-});
+const { getMessageChoices } = require('../utils/defaultMessages');
+
+let s3;
+let s3storage;
+
+if (process.env === 'production') {
```
Comment (gnorbsl) [inline server/routes/wishCards.js:50]: can you change the check to LOCAL_DEVELOPMENT so that s3 is still used on the dev server

Comment (gnorbsl) [reply server/routes/wishCards.js:50]: aside from that process.env will never be production, process.env.NODE_ENV

Comment (gnorbsl) [reply server/routes/wishCards.js:50]: for that we have USE_AWS too, so this one is not really neccessary anyway

File: server/routes/wishCards.js [hunk 2]
```diff
@@ -80,89 +102,102 @@ const AgencyRepository = require('../db/repository/AgencyRepository');
...
-        childBirthday: new Date(req.body.childBirthday),
-        wishItemPrice: Number(req.body.wishItemPrice),
-        wishCardImage: req.file.location,
-        createdBy: res.locals.user._id,
-        // Uncomment once address fields are added to profile page.
-        /* address: {
+      });
+    } else {
+      try {
+        let newWishCard = await WishCardRepository.createNewWishCard({
```
Comment (gnorbsl) [inline server/routes/wishCards.js:122]: why let if the variable is never changed again?

File: server/helper/wishcardHelper.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  if (!user) {
+    res.status(403).redirect('/users/login');
+  } else if (await IfNullOrNotVerifiedAgency(userInfo)) {
+    res.status(403).redirect('/users/profile');
+  } else {
+    next();
+  }
+};
+
+const renderPermissions = async (req, res, next) => {
```
Comment (gnorbsl) [inline server/helper/wishcardHelper.js:39]: code is almost identical; im sure there is a way to improve that

File: server/routes/wishCards.js [hunk 4]
```diff
@@ -80,89 +102,102 @@ const AgencyRepository = require('../db/repository/AgencyRepository');
...
+  },
+);
 
 // @desc    Render wishCards.html which will show all wishcards
 // @route   GET '/wishcards'
-// @access  Public, all users can see
+// @access  Private, all users can see
 // @tested 	Yes
-router.get('/', async (_req, res) => {
+router.get('/', redirectLogin, async (_req, res) => {
```
Comment (gnorbsl) [inline server/routes/wishCards.js:200]: doesnt this prevent everyone from seeing wishcards who is not logged in

**Review by gnorbsl — CHANGES_REQUESTED** [review]

File: server/routes/wishCards.js (hunk 2 above)
Comment (Jacobjeevan) [reply server/routes/wishCards.js]: Changed it to const :+1:

File: server/routes/wishCards.js (hunk 4 above)
Comment (Jacobjeevan) [reply server/routes/wishCards.js]: Yeah, true. Changed that back; changed tests as well.

File: server/helper/wishcardHelper.js (hunk 3 above)
Comment (Jacobjeevan) [reply server/helper/wishcardHelper.js:39]: Don't have any ideas as of now; any pointers? :thinking:

**Comment by Enubia** [comment]
```js
const permissionHandler = (redirect = false) => {
  return (req, res, next) => {
    const { user, agency } = req.session;
    const userInfo = { user, agency };
    if (redirect) {
      if (!user) {
        res.status(403).redirect('/users/login');
      } else if (await IfNullOrNotVerifiedAgency(userInfo)) {
        res.status(403).redirect('/users/profile');
      } else {
        next();
      }
    } else {
      if (!user) {
        res.status(403).send({ success: false, url: '/users/login' });
      } else if (await IfNullOrNotVerifiedAgency(userInfo)) {
        res.status(403).send({ success: false, url: '/users/profile' });
      } else {
        next();
      }
    }
  }
}
```

**Pull request merged**